frontend/src/components/Clock/index.js

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+
+class Clock extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { date: new Date() };
+    }
+
+    componentDidMount() {
+        this.timerID = setInterval(
+            () => this.tick(),
+            1000
+        );
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timerID);
+    }
+
+    tick() {
+        this.setState({
+            date: new Date()
+        });
+    }
+
+    render() {
+        return (
+            <div style={{ backgroundColor: 'black', color: '#ff7979' }}>
+                <h1 style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>Hello, world!</h1>
+                <h2 style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>It is {this.state.date.toLocaleTimeString()}.</h2>
+            </div>
+        );
+    }
+}
```

Comment:
this is a fairly complex one but this is how one would write this in a function component. It should work and behave the same way

```
import React, { useState, useRef, useEffect } from "react";

// one thing to note is that every time a state or a props change, the component re-render
// this means that this function run again
const Clock = () => {
  const timerIDRef = useRef(); // a ref is something that will persist between every render
  const [date, setDate] = useState(new Date());

  useEffect(() => {
    const tick = () => {
      setDate(new Date());
    };
    timerIDRef.current = setInterval(() => tick(), 1000);
    return () => {
      // cleanup function
      clearInterval(timerIDRef);
    };
  }, []);

  return (
    <div style={{ backgroundColor: "black", color: "#ff7979" }}>
      <h1 style={{ display: "flex", justifyContent: "center", alignItems: "center" }}>Hello, world!</h1>
      <h2 style={{ display: "flex", justifyContent: "center", alignItems: "center" }}>
        It is {date.toLocaleTimeString()}.
      </h2>
    </div>
  );
};
```

Reply:
Updated my Clock function to this.  Will remake the Toggle function to be in the style of this code.